edgexfoundry/edgex-go
# refactor!: Replace internal topics from config with new constants

BREAKING CHANGE: Internal topics no longer configurable, except the base topic.

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

TBD

## Testing Instructions
1. run `make docker` do build local images
2. from compose builder run `make run no-secty ds-virtual dev mqtt-broker`
3. Using MQTT tool verify external MQTT commands work
   - Subscribe to `edgex/command/response/#`
   - Publish the following to `edgex/command/request/Random-Integer-Device/Int16/get`
    ```
    {
       "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
       "apiVersion": "v2",
       "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
       "ContentType": "application/json",
       "queryParams": {
          "ds-pushevent":"true",
          "ds-returnevent":"true"
        }
    }
    ```
   - Verify valid response is received on subscribed topic.
   
4. Verify events are recieved by Core Data.
   - Enable DEBUG logging for Core Data
   - Verify logs contain messages like:
   ```
   level=DEBUG ts=2023-02-14T22:27:05.1572658Z app=core-data source=subscriber.go:65 msg="Event received from MessageBus. Topic: edgex/events/device/#, Correlation-id: 6c79cbdf-1638-4a9a-b081-afacd3673de1 "
   ```

## New Dependency Instructions (If applicable)
